### DISCUSSION TITLE
How do I do `noqa:XXX` for a specific block?

### DISCUSSION THREAD

**Discussion by smurfix** [astral-sh/ruff#8219] [Q&A]
Pylint vs. ruff:
```
class foo:
    def extreme_hack(x):  # expect no warnings
        # noqa:SLF001  # doesn't do a thing
        # pylint:disable=protected-access
        a = x._y._z
        b = x._w._v
        return a+b
    def no_hack(x):  # expect a ton of warnings
        c = x._y._z
        d = x._w._v
        return c-d
```
i.e. disable a check for the rest of the **block**, without affecting the entire module (as `ruff:noqa:…` would do).

Can ruff really not do this, or did I manage to not figure out how?

**Comment by dhruvmanila** [comment]
> Can ruff really not do this, or did I manage to not figure out how?

Ruff currently doesn't have support for block `noqa` comments (https://github.com/astral-sh/ruff/issues/3711) nor the preceding line (https://github.com/astral-sh/ruff/issues/2999).
